client/network/src/protocol.rs

```diff
@@ -223,6 +223,8 @@ pub struct Protocol<B: BlockT, H: ExHashT> {
 	protocol_engine_by_name: HashMap<Cow<'static, [u8]>, ConsensusEngineId>,
 	/// Prometheus metrics.
 	metrics: Option<Metrics>,
+	/// The `PeerId`'s of all boot nodes.
+	boot_node_ids: Arc<HashSet<PeerId>>,
```

Comment:
I would have probably just copied everything rather than sharing with `Arc`.

Reply:
We already use enough memory :P